src/net/thread.rs

```diff
@@ -0,0 +1,103 @@
+use log::{error, info};
+use std::sync::{mpsc, Arc, Mutex};
+use std::thread;
+
+use crate::error::Res;
+
+pub struct Thread {
+    worker: Worker,
+    sender: mpsc::Sender<Message>,
+}
+
+struct Worker {
+    id: usize,
```

Comment:
Did you know that you can use `(thread as JoinHandle).thread().id().as_u64()` ?

Reply:
Yes! Although there seems to be an active issue with the use of thread ID:
https://github.com/rust-lang/rust/issues/67939

Reply:
Ah, I should have checked for that.  Do we really care about this being anything other than `ThreadId`?  We don't need an integer, we just need something that implements `Debug`, don't we?

Reply:
You are right. This is only for debugging purpose, at least for now. I'll implement Debug for Worker in the next commit.